engine/org.eclipse.birt.report.engine.emitter.html/src/org/eclipse/birt/report/engine/emitter/html/HTMLEmitter.java

```diff
@@ -316,11 +318,74 @@ protected IStyle getElementStyle(IContent content) {
 		if (style == null || style.isEmpty()) {
 			return null;
 		}
+		style = inheritBorderStyle(style, content.getComputedStyle());
+		return style;
+	}
+
+	/**
+	 * The border style needs inherit properties based on the designer configuration
+	 * The style information will be transferred to the style if it set on designer
+	 * but unset at style object
+	 *
+	 */
+	private IStyle inheritBorderStyle(IStyle style, IStyle computedStyle) {
+
+		// properties: border top
+		if (computedStyle.getBorderTopStyle() != null && !computedStyle.getBorderTopStyle().equals(BORDER_STYLE_NONE)) {
```

Comment:
These days Object.equals is a nice way to detail with all the extra null guards.  Of course in this case you can compare to BORDER_STYLE_NONE which is not null.